client/my-sites/checkout/src/lib/pending-page.ts

```diff
@@ -255,6 +271,8 @@ function isRedirectAllowed( url: string, siteSlug: string | undefined ): boolean
 		'cloud.jetpack.com',
 		'jetpack.com',
 		'akismet.com',
+		'agencies.automattic.com',
+		'agencies.localhost',
```

Comment:
Since we are using an absolute path, we need to add this check to recognize valid A4A domains.